src/mpfao/tdymesh.c

```diff
@@ -963,16 +948,20 @@ static PetscErrorCode IdentifyLocalCells(DM dm, TDyMesh *mesh) {
   ierr = VecDestroy(&junkVec); CHKERRQ(ierr);
 
   ierr = DMPlexGetHeightStratum(dm, 0, &cStart, &cEnd); CHKERRQ(ierr);
+
+  PetscInt num_cells_local = 0;
   for (c=cStart; c<cEnd; c++) {
     ierr = DMPlexGetPointGlobal(dm,c,&gref,&junkInt); CHKERRQ(ierr);
     if (gref>=0) {
       cells->is_local[c] = PETSC_TRUE;
       cells->global_id[c] = gref;
+      num_cells_local++;
     } else {
       cells->is_local[c] = PETSC_FALSE;
       cells->global_id[c] = -gref-1;
     }
   }
+  mesh->num_cells_local = num_cells_local;
```

Comment:
According to this definition of "local cell", `mesh->num_cells_local != (cEnd - cStart)`. This is probably an important clue to what's going on in #224.